src/main/kotlin/org/rust/ide/navigation/goto/RsTargetElementEvaluator.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.navigation.goto
+
+import com.intellij.codeInsight.TargetElementEvaluatorEx2
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiReference
+import org.rust.lang.core.macros.ExpansionResult
+import org.rust.lang.core.macros.expandedFromRecursively
+
+/** This handler allows to intercept platform calls to [PsiReference.resolve] */
```

Comment:
Really this class doesn't intercept `PsiReference.resolve`, it only returns custom element for "goto declaration" action sometimes. Let's change this doc according to real purpose of the class